src/StateManager.ts

```diff
@@ -270,11 +270,114 @@ export default class StateManager {
                 listening: false,
             }));
         }
-
         // Draw the grid
         StateManager._gridLayer.batchDraw();
     }
 
+
+    //diagram bounds to know which exact part of the canvas to export
+    private static getDiagramBounds() {
+        if (!StateManager._nodeWrappers.length && !StateManager._transitionWrappers.length) {
+            return { x: 0, y: 0, width: 0, height: 0 };
+        }
+        let minX = Infinity, minY = Infinity, maxX = -Infinity, maxY = -Infinity;
+
+        StateManager._nodeWrappers.forEach((node) => {
+            const { x, y } = node.nodeGroup.position();
+            const radius = NodeWrapper.NodeRadius;
+            minX = Math.min(minX, x - radius);
+            minY = Math.min(minY, y - radius);
+            maxX = Math.max(maxX, x + radius);
+            maxY = Math.max(maxY, y + radius);
+        });
+
+        StateManager._transitionWrappers.forEach((transition) => {
```

Comment:
Is it necessary to check the source and destination nodes for the transition wrappers? I think all nodes that could be the source or destination for a transition would already have been iterated over within the node wrapper array. (If there is a reason to do this iteration over the transitions, though, we can certainly keep it!)

Reply:
Ou yes, you are right. xD Those are not needed, I am gonna remove them. :)